File: Tests/AINotebookCoreTests/NotebookStoreNotesTests.swift
import XCTest
@testable import AINotebookCore

@MainActor
final class NotebookStoreNotesTests: XCTestCase {

    func testCreateAndListNotes() throws {
        let store = try NotebookStore(path: .inMemory)
        let nb = try store.createNotebook(name: "NB")
        let n1 = try store.createNote(
            notebookId: nb.id!, title: "First", bodyMd: "Hello"
        )
        _ = try store.createNote(
            notebookId: nb.id!, title: "Second", bodyMd: "World"
        )
        let list = try store.notes(notebookId: nb.id!)
        XCTAssertEqual(list.count, 2)
        XCTAssertEqual(Set(list.map(\.title)), ["First", "Second"])
        XCTAssertEqual(n1.origin, .manual)
    }

    func testUpdateNoteBumpsUpdatedAt() throws {
        let store = try NotebookStore(path: .inMemory)
        let nb = try store.createNotebook(name: "NB")
        let n = try store.createNote(notebookId: nb.id!, title: "T", bodyMd: "v1")
        let originalUpdated = n.updatedAt
        Thread.sleep(forTimeInterval: 0.01)
        try store.updateNote(id: n.id!, title: "T2", bodyMd: "v2")
        let reloaded = try XCTUnwrap(store.note(id: n.id!))
        XCTAssertEqual(reloaded.title, "T2")
        XCTAssertEqual(reloaded.bodyMd, "v2")
        XCTAssertGreaterThan(reloaded.updatedAt, originalUpdated)
    }

    func testDeleteNote() throws {
        let store = try NotebookStore(path: .inMemory)
        let nb = try store.createNotebook(name: "NB")
        let n = try store.createNote(notebookId: nb.id!, title: "T", bodyMd: "x")
        try store.deleteNote(id: n.id!)
        XCTAssertNil(try store.note(id: n.id!))
    }

    func testCreateNoteWithTransformationOriginPreservesRef() throws {
        let store = try NotebookStore(path: .inMemory)
        let nb = try store.createNotebook(name: "NB")
        let n = try store.createNote(
            notebookId: nb.id!,
            title: "T",
            bodyMd: "x",
            origin: .transformation,
            originRef: 999
        )
        XCTAssertEqual(n.origin, .transformation)
        XCTAssertEqual(n.originRef, 999)
        let reloaded = try XCTUnwrap(store.note(id: n.id!))
        XCTAssertEqual(reloaded.originRef, 999)
    }
}

File: windows/tests/AINotebook.Core.Tests/Storage/NotebookStoreNotesTests.cs
using AINotebook.Core.Models;
using AINotebook.Core.Storage;
using Xunit;

namespace AINotebook.Core.Tests.Storage;

public class NotebookStoreNotesTests
{
    private static (NotebookStore store, long nb) Fresh()
    {
        var store = new NotebookStore(StorePath.InMemory);
        return (store, store.CreateNotebook("N").Id!.Value);
    }

    [Fact]
    public void CreateNoteDefaultsToManualAndGetsLowercased36CharUuid()
    {
        var (store, nb) = Fresh();
        using (store)
        {
            var note = store.CreateNote(nb, "  Title  ", "body");
            Assert.Equal(NoteOrigin.Manual, note.Origin);
            Assert.Equal("Title", note.Title);
            Assert.Equal(36, note.NoteUuid.Length);
            Assert.Contains("-", note.NoteUuid);
            Assert.Equal(note.NoteUuid.ToLowerInvariant(), note.NoteUuid);
            Assert.Null(note.AutoSourceId);
        }
    }

    [Fact]
    public void NotesOrderedByUpdatedAtDesc()
    {
        var (store, nb) = Fresh();
        using (store)
        {
            store.CreateNote(nb, "a", "x");
            var b = store.CreateNote(nb, "b", "x");
            Assert.Equal(b.Id, store.Notes(nb)[0].Id);
        }
    }

    [Fact]
    public void UpdateNoteTrimsTitleSetsBodyAndBumpsUpdatedAt()
    {
        var (store, nb) = Fresh();
        using (store)
        {
            var note = store.CreateNote(nb, "t", "v1");
            var before = note.UpdatedAt;
            store.UpdateNote(note.Id!.Value, "  t2 ", "v2");
            var reloaded = store.Note(note.Id!.Value)!;
            Assert.Equal("t2", reloaded.Title);
            Assert.Equal("v2", reloaded.BodyMd);
            Assert.True(reloaded.UpdatedAt >= before);
        }
    }

    [Fact]
    public void TransformationOriginWithOriginRefPersists()
    {
        var (store, nb) = Fresh();
        using (store)
        {
            var note = store.CreateNote(nb, "t", "b", NoteOrigin.Transformation, originRef: 999);
            var reloaded = store.Note(note.Id!.Value)!;
            Assert.Equal(NoteOrigin.Transformation, reloaded.Origin);
            Assert.Equal(999, reloaded.OriginRef);
        }
    }

    [Fact]
    public void DeleteNoteReturnsUuidAndRemoves()
    {
        var (store, nb) = Fresh();
        using (store)
        {
            var note = store.CreateNote(nb, "t", "b");
            var uuid = store.DeleteNote(note.Id!.Value);
            Assert.Equal(note.NoteUuid, uuid);
            Assert.Empty(store.Notes(nb));
        }
    }
}
